pkg/proxy/metrics/metrics.go

```diff
@@ -32,47 +32,51 @@ const kubeProxySubsystem = "kubeproxy"
 var (
 	// SyncProxyRulesLatency is the latency of one round of kube-proxy syncing proxy
 	// rules. (With the iptables proxy, this includes both full and partial syncs.)
-	SyncProxyRulesLatency = metrics.NewHistogram(
+	SyncProxyRulesLatency = metrics.NewHistogramVec(
 		&metrics.HistogramOpts{
 			Subsystem:      kubeProxySubsystem,
 			Name:           "sync_proxy_rules_duration_seconds",
 			Help:           "SyncProxyRules latency in seconds",
 			Buckets:        metrics.ExponentialBuckets(0.001, 2, 15),
 			StabilityLevel: metrics.ALPHA,
 		},
+		[]string{"ipFamily"},
```

Comment:
additive labels is not a breaking change

Reply:
how could it not be? people would be expecting a scalar value and would now get back a map or whatever (or need to specify a label that they didn't know they needed to specify)

Reply:
Adding a label is considered a breaking change: https://github.com/kubernetes/community/blob/54377a12a6fa345c15b21be278a57efe6296304e/contributors/devel/sig-instrumentation/metric-stability.md#stable

It breaks the assumptions that users have towards the metric, but it is not going to break the users' monitoring platforms unless the dimension you are trying to add is unbounded.

The concern about going from a scalar to a vector is understandable, but as far as I am aware this should never break users' platform because operations on scalar are compatible with vectors, but the opposite is not true.

The takeaway is that this is a breaking change for Beta / Stable metrics which requires going through a deprecation cycle, but for an Alpha metric this is expected and normal.

Reply:
> It breaks the assumptions that users have towards the metric, but it is not going to break the users' monitoring platforms unless the dimension you are trying to add is unbounded.

this, everything that worked before will keep working